src/contract/contractmanager.cpp

```diff
@@ -14,10 +14,8 @@ See the LICENSE.txt file in the project root for more information.
 #include "../core/state.h"
 #include "../utils/dynamicexception.h"
 
-ContractManager::ContractManager(
-  DB& db, State& state,
-  rdPoS& rdpos, const Options& options
-) : BaseContract("ContractManager", ProtocolContractAddresses.at("ContractManager"), options.getChainOwner(), 0, db),
+ContractManager::ContractManager(DB& db, State& state, rdPoS& rdpos, const Options& options)
+: BaseContract("ContractManager", ProtocolContractAddresses.at("ContractManager"), options.getChainOwner(), 0, db),
```

Comment:
I know you just changed the indentation here, but in any case, what about removing the separate rdPoS& argument and just using a e.g. state.getRdPoS() / state.getConsensus() where the rdPoS& is needed?

Reply:
Same as the previous comment, but this specific case is an exception - ContractManager was allowed to keep rdPoS& as an argument because it also exists within and is owned exclusively by State. They're on the same hierarchy level, so it's not necessary for CM to go through State to call rdPoS.